src/reporter/utils/pypi.py

```diff
@@ -0,0 +1,14 @@
+"""Utilities related to PyPI"""
+
+from urllib.parse import urlparse
+
+
+def file_path_from_inspector_url(inspector_url: str) -> str:
+    """Parse the file path out of a PyPI inspector URL"""
+
+    parsed_url = urlparse(inspector_url)
+    path = parsed_url.path.strip("/")
+    segments = path.split("/")
+
+    # The 8th element of path is when the file path starts
+    return "/".join(segments[8:])
```

Comment:
Is it always going to be the 8th element? I dunno, this feels kinda hacky, but it's just a nit.

Reply:
> Is it always going to be the 8th element? I dunno, this feels kinda hacky, but it's just a nit.

It should always be, but we're also passing the path parameter down elsewhere IIRC. I don't see a reason that can't make its way back up to the reporter API in a safe way, though it might motivate a larger code change.

Reply:
The path isn't passed up from anywhere - it's converted to an inspector_url on the scanner client itself and that's propagated upwards, so that's all we have to work with unfortunately.

Reply:
> The path isn't passed up from anywhere - it's converted to an inspector_url on the scanner client itself and that's propagated upwards, so that's all we have to work with unfortunately.

That's... not entirely true is it? This might require a schema change, but objectively the first 'working instance' of a path we receive is via the PyPI JSON API response. We construct the Inspector URL from the path to the pythonhosted resource.

Objectively we _do_ have a better way to do this; we simply retain the path that we originally received from the PyPI JSON API response in our backend. But I don't know that it's any better of a solution.

Reply:
Sorry that was unclear, yes it's possible and probably the right way to do it, I'm just saying we don't currently actually have it implemented. There's no database field for "path" (only `inspector_url`). If this is something you want to see implemented it could be done in a separately scoped issue/PR

Reply:
> Sorry that was unclear, yes it's possible and probably the right way to do it, I'm just saying we don't currently actually have it implemented. There's no database field for "path" (only `inspector_url`). If this is something you want to see implemented it could be done in a separately scoped issue/PR

Frankly that might be more useful if we have to wholesale reference any specific packages. But I'll leave it up to you to decide. I agree with Aboo that this isn't an excellent way to do it, but I also don't see that schema changing in any way that is particularly harmful to what we're doing. Except that I think this will throw an error if the report modal is supplied less than the required number of slashes. (E.g., I mistype and input the package at its root without the hash.) 
`https://inspector.pypi.io/project/numpy/1.26.4/`
and 
`https://inspector.pypi.io/project/numpy/`
are both valid URL's.

Reply:
Yes that will unfortunately throw an error. I think for now it's fine to leave as-is but in the long run I do have some ideas for a better schema that will be better for us.